nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -253,7 +278,13 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                     });
 
                     outgoingFlowFile = session.putAllAttributes(outgoingFlowFile, attributes);
-                    session.getProvenanceReporter().receive(outgoingFlowFile, getURI(context));
+                    String uriPass = "";
+                    if (context.getProperty(USER_NAME).getValue() != null) {
+                        uriPass = "mongodb://" + context.getProperty(USER_NAME).getValue() + ":" + context.getProperty(PASSWORD).getValue() + "@" + getURI(context).substring(10);
```

Comment:
I don't believe the URI is validated anywhere to ensure it starts with `mongodb://`, so arbitrarily starting the index at _10_ isn't guaranteed to produce the expected outcome.

Reply:
Also, there is a note in the [MongoDB docs](https://docs.mongodb.com/manual/reference/connection-string/) regarding usernames and passwords that contain special characters: 

> If the username or password includes the at sign @, colon :, slash /, or the percent sign % character, use [percent encoding](https://tools.ietf.org/html/rfc3986#section-2.1).

Reply:
I will add in the validation for the starting 10 characters for the URI. 

I did see the MongoDB doc, for percent encoding which would be better:
a) forcing the user to handle the percent encoding and printing it in the description as a note for user.
b) explicitly handling of it by the processor. 

current implementation is goes the first way. But if you think It is better for processor to handle, I will look into it.

Reply:
This is not something I would expect the user to do manually. It should be done automatically by the processor.